File: Bubble Sort.py
# REPEAT number of items in the list -1 times
#     FOR each adjacent pair of items in the list
#         IF first of the pair > second of pair THEN 
#             Swap positions 
#         END IF 
#     NEXT pair of Items
# END REPEAT

def BubbleSort(list_a):   #this is an enhanced alternative bubble sort using a Boolean value.
    indexing_length = len(list_a) - 1
    sorted = False

    while not sorted:
        sorted = True
        for i in range(0, indexing_length):
            if list_a[i] > list_a[i + 1]:
                sorted = False
                list_a[i], list_a[i + 1] = list_a[i + 1], list_a[i]
        return list_a

list_a = [9, 12, 4, 3]

a = BubbleSort(list_a)

print(a)

File: bubble_sort.c
#include<stdio.h>

int sort(int array[], int length) 
{
    int temp, i, j;

    for(i = 0; i < length; i++)
    {
        for(j = 0; j < (length - 1); j++)
        {
            if(array[j]>array[j + 1])
            {
                temp = array[j];
                array[j] = array[j + 1];
                array[j + 1] = temp;
            }
        }
    }

    printf("\n");
    
    for(i = 0; i < length; i++)
    {
        printf("array[%d] = %d\n", i, array[i]);
    }
}

int main() 
{
    int length, i, sorted;

    printf("enter length of the array: ");
    scanf("%d", &length); 

    int array[length];

    for(i = 0; i < length; i++)
    {
        printf("array[%d] = ", i);
        scanf("%d", &array[i]);
    }

    sort(array, length);

    return 0;
}